cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ExecutionTaskPlanner.java

```diff
@@ -51,18 +50,24 @@
   private final Set<ExecutionTask> _remainingReplicaMovements;
   private final Map<Long, ExecutionTask> _remainingLeadershipMovements;
   private long _executionId;
-  private final ExecutionTaskStrategy _defaultExecutionStrategy;
+  private ReplicaMovementStrategy _replicaMovementTaskStrategy;
 
-  public ExecutionTaskPlanner(String defaultExecutionStrategy) {
+  public ExecutionTaskPlanner(List<String> replicaMovementStrategies) {
     _executionId = 0L;
     _partMoveTaskByBrokerId = new HashMap<>();
     _remainingReplicaMovements = new TreeSet<>();
     _remainingDataToMove = 0L;
     _remainingLeadershipMovements = new HashMap<>();
-    try {
-      _defaultExecutionStrategy = (ExecutionTaskStrategy) Class.forName(defaultExecutionStrategy).newInstance();
-    } catch (Exception e) {
-      throw new IllegalStateException("Default execution strategy " + defaultExecutionStrategy + " is undefined.");
+    for (String replicaMovementStrategy : replicaMovementStrategies) {
+      try {
+        if (_replicaMovementTaskStrategy == null) {
+          _replicaMovementTaskStrategy = (ReplicaMovementStrategy) Class.forName(replicaMovementStrategy).newInstance();
+        } else {
+          _replicaMovementTaskStrategy = _replicaMovementTaskStrategy.chain((ReplicaMovementStrategy) Class.forName(replicaMovementStrategy).newInstance());
+        }
+      } catch (Exception e) {
+        throw new RuntimeException(e);
```

Comment:
Nit: Would it be more clear to add a message in addition to the stack trace? -- i.e. 
```
throw new RuntimeException("Error occurred while setting up the replica movement strategy: " + replicaMovementStrategy + ".", e);
```